lib/inferno/apps/cli/evaluate.rb

```diff
@@ -2,13 +2,58 @@
 require_relative '../../dsl/fhir_evaluation/config'
 require_relative '../../entities'
 require_relative '../../utils/ig_downloader'
+require_relative 'migration'
 
 require 'tempfile'
+require 'faraday'
 
 module Inferno
   module CLI
-    class Evaluate < Thor::Group
-      def evaluate(ig_path, data_path, _log_level)
+    class Evaluate
+      # @see Inferno::CLI::Main#evaluate
+      def run(ig_path, data_path, options)
+        tmpdir = Dir.mktmpdir
+        Dir.mkdir("#{tmpdir}/data")
+        Dir.mkdir("#{tmpdir}/data/igs")
+        ENV['TMPDIR'] = tmpdir
+        ENV['FHIRPATH_URL'] = 'http://localhost:6789'
+        ENV['FHIR_RESOURCE_VALIDATOR_URL'] = 'http://localhost:3500'
+
+        puts 'Starting Inferno Evaluator Services...'
+        system("#{services_base_command} up -d #{services_names}")
+
+        ig_path = if ig_path.starts_with? '~'
+                    ig_path.sub('~', Dir.home)
+                  else
+                    File.absolute_path(ig_path)
+                  end
+
+        Dir.chdir(tmpdir) do
+          Migration.new.run(Logger::FATAL) # Hide migration output for evaluator
+          evaluate(ig_path, data_path, options)
+        end
+      ensure
+        system("#{services_base_command} down #{services_names}")
+        puts 'Stopped Inferno Evaluator Services'
+
+        FileUtils.remove_entry_secure tmpdir
+      end
+
+      def services_base_command
+        "docker compose -f #{File.join(__dir__, 'evaluate', 'docker-compose.evaluate.yml')}"
+      end
+
+      def services_names
+        return @service_names if @service_names
+
+        fhirpath_health = Faraday.get("#{ENV.fetch('FHIRPATH_URL', nil)}/version").status
+        @service_names = "hl7_validator_service #{fhirpath_health == 200 ? '' : 'fhirpath'}"
```

Comment:
Why is this logic only needed for the fhirpath service? I would expect it to be present for the validator service as well, or just a plain `docker compose up` without the service names.

Reply:
The validator service doesn't expose its port so it doesn't conflict with spinning it up for the evaluator. On the other hand if we wanted the evaluator to leverage an already running validator service we would have to route it through Nginx or modify test kits to expose the validator port as well.

Reply:
But the docker-compose file here is mapping both services to different ports, so neither should conflict.